biloba/service.py

```diff
@@ -143,7 +139,7 @@ def stop(self, block=True):
         if not self.started:
             return
 
-        self._kill.set()
+        self.pool.kill(block=block)
```

Comment:
In fact according to https://github.com/gevent/gevent/blob/master/gevent/pool.py#L101 - block is ``True`` by default - this should explictly be called as ``block=False`` here.